### PR TITLE
Fix, replace for..of on for loop

### PR DESCRIPTION
**Pull request by NikitaDudin**
## Description
Application crash when using transform or color properties on a style object. Replacing for..of loop on for loop solves it problem.

### Issue
- #2120
- #2125

### CODE REVIEW

**Comment by piaskowyk** [comment]
Could you provide an example code to reproduce this issue?

**Comment by NikitaDudin** [comment]
@piaskowyk https://github.com/software-mansion/react-native-reanimated/pull/1879#issuecomment-853412859

**Comment by mrousavy** [comment]
Looks like the JS runtime (or the babel plugin) doesn't whitelist the `iterator` as a native prop. `for..of` uses `[Symbol.iterator]`, no?

File: src/reanimated2/Hooks.ts [hunk 1]
```diff
@@ -162,7 +162,8 @@ function runAnimations(animation, timestamp, key, result, animationsActive) {
 function isAnimated(prop) {
   'worklet';
   if (Array.isArray(prop)) {
-    for (const item of prop) {
+    for (let i = 0; i < prop.length; i += 1) {
+      const item = prop[i]
```
Comment (terrysahaidak) [inline src/reanimated2/Hooks.ts:166]:
```suggestion
      const item = prop[i];
```

**Review by piaskowyk — CHANGES_REQUESTED** [review]
Thanks for PR 🙌
Could you just change `i += 1` to `++i`?

**Comment by piaskowyk** [comment]
@NikitaDudin which version of babel and Hermes do you use?

**Comment by NikitaDudin** [comment]
@piaskowyk hermes: 0.7.2 babel: 7.9.0

**Comment by NikitaDudin** [comment]
> Thanks for PR 🙌
> Could you just change `i += 1` to `++i`?

Are you sure you should change to the prefix form `++i`?  May be `i++`?

**Comment by piaskowyk** [comment]
> > Thanks for PR 🙌
> > Could you just change `i += 1` to `++i`?
> 
> Are you sure me should change to the prefix form `++i`? May be `i++`?

Yes, `++i` is a little faster than `i++` and `i++` is faster than `i += 1` 😅. Sure it can depend on the runtime version, but general idea is that. I think that every above expression is the same readable, but if we can choose faster - why not - what do you think? 🤔

**Comment by piaskowyk** [comment]
I generally don't pick on things like that because this doesn't make a big difference, but I optimized it last time and I want to make it as faster as possible. 😅

**Review by piaskowyk — APPROVED** [review]

**Comment by piaskowyk** [comment]
@NikitaDudin good job, thanks 💪

**Pull request merged**